src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -300,7 +300,13 @@ public List<ChainedHeader> PartialValidationSucceeded(ChainedHeader chainedHeade
 
             if (!this.chainedHeadersByHash.ContainsKey(chainedHeader.HashBlock))
             {
-                this.logger.LogTrace("(-)[HEADER_NOT_FOUND]");
+                this.logger.LogTrace("(-)[HEADER_NOT_FOUND]:null");
+                return null;
+            }
+
+            if (chainedHeader.Block == null)
```

Comment:
This can happen?
In what cases?

Reply:
Really odd race condition. Very unlikely to happen.

Reply:
We will add comments for those new guys to explain why can it happen